File: FsharpTodoApp.Infrastructure/Persistence/AppDbContext.fs
namespace FsharpTodoApp.Infrastructure.Persistence

open Microsoft.EntityFrameworkCore
open FsharpTodoApp.Infrastructure.Features.Todo.DataModels
open FsharpTodoApp.Infrastructure.Features.User.DataModels

type AppDbContext(options) =
    inherit DbContext(options)

    override _.OnConfiguring optionsBuilder =
        optionsBuilder.UseQueryTrackingBehavior QueryTrackingBehavior.NoTracking
        |> ignore

        base.OnConfiguring optionsBuilder

    [<DefaultValue>]
    val mutable Todos: DbSet<TodoDataModel>

    [<DefaultValue>]
    val mutable Users: DbSet<UserDataModel>

    override _.OnModelCreating modelBuilder =
        TodoDataModel.onModelCreating modelBuilder
        UserDataModel.onModelCreating modelBuilder
        base.OnModelCreating modelBuilder

File: FsharpTodoApp.Persistence/AppDbContext.cs
using FsharpTodoApp.Persistence.DataModels;
using Microsoft.EntityFrameworkCore;

namespace FsharpTodoApp.Persistence;

public class AppDbContext(DbContextOptions<AppDbContext> options) : DbContext(options)
{
    protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
    {
        optionsBuilder.UseQueryTrackingBehavior(QueryTrackingBehavior.NoTracking);
        base.OnConfiguring(optionsBuilder);
    }

    public DbSet<TodoDataModel> Todos { get; set; } = null!;
    public DbSet<UserDataModel> Users { get; set; } = null!;

    protected override void OnModelCreating(ModelBuilder modelBuilder)
    {
        TodoDataModel.OnModelCreating(modelBuilder);
        UserDataModel.OnModelCreating(modelBuilder);

        base.OnModelCreating(modelBuilder);
    }
}
